### ISSUE TITLE
[naming-convention] allow empty prefix value

### ISSUE THREAD

**Issue by criskrzysiu** [typescript-eslint/typescript-eslint#2795]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/naming-convention": [
        "error",
        {
          "selector": "enumMember",
          "format": ["PascalCase"],
          "prefix": ["", "is"]
        }
      ]
  }
}
```

```TS
// your repro code case
```

**Expected Result**

I want to allow prefix "is" or no prefix for enum member;

So both "Error" and "isError" for example will be allowed, but "error" will throw error;

**Actual Result**
```
Configuration for rule "@typescript-eslint/naming-convention" is invalid:
        Value "" should NOT be shorter than 1 characters.
        Value "" should NOT be shorter than 1 characters.
        Value "" should NOT be shorter than 1 characters.
        Value "" should NOT be shorter than 1 characters.
        Value "" should NOT be shorter than 1 characters.
        Value "" should NOT be shorter than 1 characters.
        Value "" should NOT be shorter than 1 characters.
        Value "" should NOT be shorter than 1 characters.
        Value "" should NOT be shorter than 1 characters.
        Value "" should NOT be shorter than 1 characters.
        Value "" should NOT be shorter than 1 characters.
        Value "" should NOT be shorter than 1 characters.
        Value "" should NOT be shorter than 1 characters.
        Value "" should NOT be shorter than 1 characters.
        Value "" should NOT be shorter than 1 characters.
        Value "" should NOT be shorter than 1 characters.
        Value "" should NOT be shorter than 1 characters.
        Value "" should NOT be shorter than 1 characters.
        Value {"selector":"enumMember","format":["PascalCase"],"prefix":["","is"]} should match exactly one schema in oneOf.
```

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.3.0` |
| `@typescript-eslint/parser`        | `4.3.0` |
| `TypeScript`                       | `4.0.5` |
| `ESLint`                           | `7.6.0` |
| `node`                             | `14.4.0` |

**Comment by bradzacher** [comment]
Why not just use the custom format?

A regex like `/(is)?[A-Z][a-zA-Z0-9]*/` will do exactly what you want.

**Comment by criskrzysiu** [comment]
yes, You can do it but it is much less clear what is going on, and writing this for multiple allowed formats will be difficult. Allowing empty prefix values is more verbose. Other option would be to add "prefixReqired" option.
 
```
{
  "rules": {
    "@typescript-eslint/naming-convention": [
        "error",
        {
          "selector": "enumMember",
          "format": ["PascalCase"],
          "prefix": ["is"],
          "prefixRequired": false
        }
      ]
  }
}
```

**Comment by bradzacher** [comment]
Given that nobody else has engaged with this - I don't think there's enough community need for it.
So I'm going to close this as not going to implement.
